_posts/2023-08-02-#28122.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: pr
+date: 2023-08-02
+title: "Silent Payments: Implement BIP352"
+pr: 28122
+authors: [josibake]
+components: ["wallet"]
+host: josibake
+status: upcoming
+commit:
+---
+
+## Notes
+
+This PR is the first step towards implementing [BIP352: Silent Payments](https://github.com/bitcoin/bips/pull/1458) in the Bitcoin Core wallet. BIP352 is "a protocol for static payment addresses in Bitcoin without on-chain linkability of payments or a need for on-chain notifications."
+
+Before reviewing the PR, it is _strongly_ recommended that you read the [BIP](https://github.com/bitcoin/bips/pull/1458), as well as the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json). This PR implements the logic from the BIP and is a child PR of [Silent Payments: send and receive](https://github.com/bitcoin/bitcoin/pull/27827).
+
+## Questions
+
+1. Did you read the [BIP](https://github.com/bitcoin/bips/pull/1458)? In a few sentences, can you summarize it in your own words?
+
+1. Did you review the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json)? Can you think of additional test cases?
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+1. In [5688262](https://github.com/bitcoin/bitcoin/pull/28122/commits/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
```

Comment:
For these and the other links, we'll want to use the review club mirror, otherwise they'll disappear as soon as you force push to the PR. I've pushed a tag at https://github.com/bitcoin-core-review-club/bitcoin/commits/pr28122

```suggestion
1. In [5688262](https://github.com/bitcoin-core-review-club/bitcoin/commit/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
```

Reply:
:+1: Is it possible to do the same for the BIPs repo? There are links to the reference.py implementation and test vectors file from the BIP pull request, those will also break once if I need to update the BIP. Alternatively, I can just have all the BIP links point just to the PR, whichever is easier.

Reply:
Ah yes no problem, I made a mirror and pushed a tag: https://github.com/bitcoin-core-review-club/bips/tree/pr28122

Feel free to use e.g. https://github.com/bitcoin-core-review-club/bips/blob/174ba63f44ddab89b73c21f90d2f5cb1968b6db1/bip-0352/bech32m.py

Reply:
Done!